### ISSUE TITLE
Reproducing Results

### ISSUE THREAD

**Issue by AkshitSharma1** [SiddhantBikram/MemeCLIP#1]
Hello!
Thank you for sharing the code and the impressive results.

While we were able to nearly reproduce for hate classification on PrideMM using the configs.py given (with the exception of having to guess cfg.ratio), the same couldn't be done for other tasks (target, stance & humour classification). Please could you share the exact config.py used for them along with the one used to produce results on Harmeme dataset as mentioned in the paper?

Thank you!

**Comment by SiddhantBikram** [comment]
Hi there! Thank you for raising this issue and for your interest in our work.

The model weights provided in README.md are for the hate classification task on PrideMM. To reproduce the results for the target, stance, and humor classification tasks, please train the model on those tasks specifically by changing cfg.label in configs.py. Between experiments in different tasks, no parameters except cfg.label were changed in configs.py. The metrics reported in the paper were obtained after training the model from scratch and averaged across 3 seeds.

Thank you for pointing out the issue with cfg.ratio! As reported in the paper, this hyperparameter was fixed to 0.2 for all experiments, and the missing value has been added in configs.py.

**Comment by AkshitSharma1** [comment]
Hi,

Thank you for the response.  

As instructed by you, the below configs.py was used for all 4 tasks - by changing the cfg.label value
```

cfg.clip_variant = "ViT-L/14"
cfg.dataset_name = 'Pride'
cfg.name = 'MemeCLIP' 
cfg.label = 'humour' #Only this value was changed based on task
cfg.seed = 42
cfg.test_only = False
cfg.device = 'cuda'
cfg.gpus = [0]

if cfg.label =='hate':
    cfg.class_names = ['Benign Meme', 'Harmful Meme']
elif cfg.label == 'humour':
    cfg.class_names = ['No Humour', 'Humour']
elif cfg.label == 'target':
    cfg.class_names = ['No particular target', 'Individual', 'Community', 'Organization']
elif cfg.label == 'stance':
    cfg.class_names = ['Neutral', 'Support', 'Oppose']
cfg.reproduce = False
cfg.batch_size = 16
cfg.image_size = 224
cfg.num_mapping_layers = 1
cfg.unmapped_dim = 768
cfg.map_dim = 1024
cfg.num_pre_output_layers = 1
cfg.drop_probs = [0.1, 0.4, 0.2]
cfg.lr = 1e-4
cfg.max_epochs = 10
cfg.ratio = 0.2
cfg.weight_decay = 1e-4
cfg.num_classes = len(cfg.class_names)
cfg.scale = 30 
cfg.print_model = True
```

With this, the below results were obtained-

1. Hate task
![Image](https://github.com/user-attachments/assets/30660eb3-ffdf-44b1-bceb-bf39a48b688c)
While those reported in the paper 
**Accuracy:** 76.06 ±0.23
**AUROC (Macro):**  84.52±0.31
**F1(Macro):** 75.09±0.20


2. Target task
![Image](https://github.com/user-attachments/assets/f9af7508-7f68-45f5-92a9-f22589c6b64c)
While those reported in the paper 
**Accuracy:** 66.12 ± 0.47
**AUROC (Macro):** 81.66± 0.25
**F1(Macro):** 58.65±0.97

3. Stance task
![Image](https://github.com/user-attachments/assets/c85698c1-d559-445f-8bd7-8ac0f9d8fe98)
While those reported in the paper 
**Accuracy:** 62.00±0.12
**AUROC (Macro):** 80.11±0.15
**F1(Macro):** 57.98±1.91

4. Humor task
![Image](https://github.com/user-attachments/assets/47597c55-c454-41fb-8125-73b2d696f7eb)
While those reported in the paper 
**Accuracy:** 80.27±0.52
**AUROC (Macro):** 85.59±0.23 
**F1(Macro):** 77.21±0.79

The results for Hate, Target and Humor task using seed 42 seem to be a bit off from the values stated in paper. Could you please let us know the other two seeds which have been used for averaging results?

Also it would be of great help if you could please share the configuration used to evaluate the model on HarMeme dataset

Thank you!

**Comment by SiddhantBikram** [comment]
MemeCLIP's performance on the PrideMM dataset was computed using the seeds 42, 100, and 510. Here are the full raw results for your reference:

Hate Classification:

|   Seed      |    Accuracy      |     AUROC      |     F1      |
|:------------:|:------------:|:-------------:|:-------------:|
| 42           | 75.93        | 84.23         | 74.98         |
| 100          | 75.93        | 84.85         | 74.97         |
| 510          | 76.33        | 84.49         | 75.33         |
| Mean  | 76.06333333  | 84.52333333   | 75.09333333   |
|  SD | 0.2309401077 | 0.3113411848   | 0.2050203242  |

Target Classification:

|    Seed      |    Accuracy     |     AUROC      |     F1      |
|:------------:|:-----------:|:-------------:|:-------------:|
| 42           | 66.39       | 81.53         | 58.45         |
| 100          | 65.58       | 81.90         | 57.79         |
| 510          | 66.39       | 81.50         | 59.70         |
| Mean  | 66.12       | 81.66333333   | 58.64666667   |
| SD  | 0.467653718 | 0.248461935   | 0.9700687261  |

Stance Classification:

|    Seed      |     Accuracy     |    AUROC     |    F1      |
|:------------:|:------------:|:------------:|:-------------:|
| 42           | 61.93        | 80.28        | 58.49         |
| 100          | 61.93        | 80.08        | 55.87         |
| 510          | 62.13        | 79.98        | 59.59         |
| Mean | 61.99666667  | 80.11333333  | 57.99         |
| SD  | 0.1154700538 | 0.1527525232 | 1.911055555   |

Direction of Hate Classification:

|    Seed      |     Accuracy      |     AUROC       |     F1      |
|:------------:|:--------------:|:---------------:|:--------------:|
| 42           | 80.47          | 85.46           | 77.39          |
| 100          | 80.67          | 85.45           | 77.90          |
| 510          | 79.68          | 85.86           | 76.34          |
| Mean |  80.27333333    | 85.59   | 77.21  |
| SD  | 0.5234819322           | 0.2338803113            | 0.795424415         |

For the HarMeme dataset, we used the same settings as the Hate Classification task for PrideMM.
